docs/architecture/adr-056-proving-amnesia-attack.md

```diff
@@ -0,0 +1,117 @@
+# ADR 056: Proving flip flopping attacks
+
+## Changelog
+
+- 02.04.20: Initial Draft
+- 06.04.20: Second Draft
+
+## Context
+
+Whilst most created evidence of malicious behaviour is self evident such that any individual can verify them independently there are types of evidence, known as global evidence, that require further collaboration from the network in order to accumulate enough information to create evidence that is individually verifiable and can therefore be processed through consensus. [Fork Accountability](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md) as a whole constitutes of detection, proving and punishing. This ADR addresses how to prove infringement from global evidence that is sent to a full node.
+
+The currently only known form of global evidence stems from [flip flopping](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md#flip-flopping) attacks. The schematic below explains one scenario where an amnesia attack, a form of flip flopping, can occur such that C1 and C2 commit different blocks.
+
+![](../imgs/tm-amnesia-attack.png)
+
+1. C1 and F send PREVOTE messages
+2. C1 sends PRECOMMIT for round 1
+3. A new round is started, C2 and F send PREVOTE messages
+4. C2 and F then send PRECOMMIT messages
+5. F breaks the lock and goes back and sends PRECOMMIT messages in round 1
+
```

Comment:
I think we should make clear that the problematic case is if different blocks are proposed in the first and in the second round. Because this also reduces the cases the PoLC actually has to be stored

Reply:
Yes good point, I will add that in